File: pkg/codegen/testing/utils/testdata/components-pp/python/another_component.py
import pulumi
from pulumi import Input
from typing import Optional, Dict, TypedDict, Any
import pulumi_random as random

class AnotherComponent(pulumi.ComponentResource):
    def __init__(self, name: str, opts: Optional[pulumi.ResourceOptions] = None):
        super().__init__("components:index:AnotherComponent", name, {}, opts)

        first_password = random.RandomPassword(f"{name}-firstPassword",
            length=16,
            special=True,
            opts = pulumi.ResourceOptions(parent=self))

        self.register_outputs()

File: pkg/codegen/dotnet/testdata/components-pp/AnotherComponent.cs
using System.Collections.Generic;
using System.Linq;
using Pulumi;
using Random = Pulumi.Random;

namespace Components
{
    public class AnotherComponent : global::Pulumi.ComponentResource
    {
        public AnotherComponent(string name, ComponentResourceOptions? opts = null)
            : base("components:index:AnotherComponent", name, ResourceArgs.Empty, opts)
        {
            var firstPassword = new Random.RandomPassword($"{name}-firstPassword", new()
            {
                Length = 16,
                Special = true,
            }, new CustomResourceOptions
            {
                Parent = this,
            });

            this.RegisterOutputs();
        }
    }
}
